drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/CallbackFactoryTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.guided.dtable.client.widget.table.columns;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.drools.workbench.screens.guided.dtable.client.widget.table.columns.dom.listbox.MultiValueDOMElement;
+import org.drools.workbench.screens.guided.dtable.client.widget.table.columns.dom.listbox.MultiValueSingletonDOMElementFactory;
+import org.gwtbootstrap3.client.ui.ListBox;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.uberfire.client.callbacks.Callback;
+import org.uberfire.ext.wires.core.grids.client.model.GridCell;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class CallbackFactoryTest {
+
+    private static final String KEY = "key";
+
+    private static final String VALUE = "value";
+
+    @Mock
+    private MultiValueSingletonDOMElementFactory factory;
+
+    @Mock
+    private MultiValueDOMElement multiValueDOMElement;
+
+    @Mock
+    private ListBox multiValueWidget;
+
+    @Mock
+    private GridCell cell;
+
+    private Map<String, String> enumLookups;
+
+    @Before
+    public void setup() {
+        enumLookups = new HashMap<>();
+        when(multiValueDOMElement.getWidget()).thenReturn(multiValueWidget);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void makeOnCreationCallbackListBoxSingleSelect() {
+        setupMultipleSelectTest(CallbackFactory.MAX_VISIBLE_ROWS,
+                                false);
+
+        final Callback callback = CallbackFactory.makeOnCreationCallback(factory,
+                                                                         cell,
+                                                                         enumLookups);
+        callback.callback(multiValueDOMElement);
+
```

Comment:
Any chance to add `verify(multiValueWidget, never()).setVisibleItemCount(anyInt());`?

Reply:
Done.

Reply:
Thanks